### PR TITLE
feat(frontend): add explorer URL for ckERC20

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

We adjust the logic of the explorer URL: up until now, the explorer was only provided with the condition that the `ledgerCanisterId` is either ckETH or ckSepoliaETH. We provide it now even for other ckTokens.

### CODE REVIEW

**Review by peterpeterparker — COMMENTED** [review]
Why the change? Can you provide some reason / example on why we need to adjust the logic? That way I could more easily assert if the code change matches the goal. Thanks in advance.

File: src/frontend/src/icp/utils/cketh-transactions.utils.ts [hunk 1]
```diff
@@ -40,14 +40,12 @@ export const mapCkEthereumTransaction = ({
...
 
-	const ckETHExplorerUrl =
-		IC_CKETH_LEDGER_CANISTER_ID === ledgerCanisterId && nonNullish(env)
-			? env === 'testnet' && nonNullish(STAGING_CKETH_LEDGER_CANISTER_ID)
-				? ledgerCanisterId !== STAGING_CKETH_LEDGER_CANISTER_ID
-					? `${CKETH_SEPOLIA_EXPLORER_URL}/${ledgerCanisterId}`
-					: CKETH_SEPOLIA_EXPLORER_URL
-				: CKETH_EXPLORER_URL
-			: undefined;
+	const ckETHExplorerUrl = nonNullish(env)
```
Comment (peterpeterparker) [inline src/frontend/src/icp/utils/cketh-transactions.utils.ts:43]: What's `+`? If the goal is to compose a string, can we use string literals similarly as in the rest of the codebase?

File: src/frontend/src/icp/workers/icrc-wallet.worker.ts [hunk 2]
```diff
@@ -50,7 +50,7 @@ const mapTransaction = ({
 	}
 
 	if (nonNullish(ledgerId) && (isTokenCkEthLedger(ledgerId) || isTokenCkErc20Ledger(ledgerId))) {
-		return mapCkEthereumTransaction({ transaction, identity, ...ledgerId });
+		return mapCkEthereumTransaction({ transaction, identity, ...env, ...ledgerId });
```
Comment (peterpeterparker) [inline src/frontend/src/icp/workers/icrc-wallet.worker.ts:53]: Why this change? Is that related to this PR?

File: src/frontend/src/icp/utils/cketh-transactions.utils.ts (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp/utils/cketh-transactions.utils.ts:43]: something seemed off when I wrote it... that was it! thank you!

File: src/frontend/src/icp/workers/icrc-wallet.worker.ts (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp/workers/icrc-wallet.worker.ts:53]: Yes, correct! Basically, `env` was not being passed to the function, making it `undefined` and consequently `ckETHExplorerUrl` was `undefined`. 

Do you want me to put it in another PR? I think it is quite related to this one, no?

Comment (peterpeterparker) [reply src/frontend/src/icp/workers/icrc-wallet.worker.ts:53]: Yes please another PR, thanks!

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp/workers/icrc-wallet.worker.ts:53]: Done in 

https://github.com/dfinity/oisy-wallet/pull/2130

**Comment by AntonioVentilii-DFINITY** [comment]
@peterpeterparker  ready for another review

File: src/frontend/src/icp/utils/cketh-transactions.utils.ts [hunk 3]
```diff
@@ -40,14 +40,13 @@ export const mapCkEthereumTransaction = ({
...
+		? `${
+				env === 'testnet' && nonNullish(STAGING_CKETH_LEDGER_CANISTER_ID)
+					? CKETH_SEPOLIA_EXPLORER_URL
+					: CKETH_EXPLORER_URL
+			}${!CKETH_LEDGER_CANISTER_IDS.includes(ledgerCanisterId) ? `/${ledgerCanisterId}` : ''}`
```
Comment (peterpeterparker) [inline src/frontend/src/icp/utils/cketh-transactions.utils.ts:48]: `isTokenCkErc20Ledger`?

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp/utils/cketh-transactions.utils.ts]: yes! thank you!

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx
